modules/virt-configuring-masquerade-mode-dual-stack.adoc

```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-configuring-masquerade-mode-dual-stack_{context}"]
+= Configuring masquerade mode with dual-stack (IPv4 and IPv6)
+
+You can configure virtual machines to use both IPv6 and IPv4 on the default Pod Network. The IPv6 network address must be statically set in the virtual machine.
+
+Once configured, incoming and outgoing traffic for the virtual machine is routed to both the IPv4 address and the unique IPv6 address of the virt-launcher Pod.
+The virt-launcher Pod then routes the IPv4 traffic to the DHCP address of the virtual machine, and the IPv6 traffic to the statically set IPv6 address of the virtual machine.
+
+This procedure can be applied to new or existing virtual machines in the cluster.
+
+.Prerequisites
+
+* The underlying network for the cluster must be connected with IPv6.
+
+* The OpenShift cluster must be configured to use IPv6.
+
+.Procedure
+
+. Edit the virtual machine configuration to add an interface with `masquerade` and configure the IPv6 IP address, default gateway, and DNS record using cloud-init.
++
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: example-vm-ipv6
+...
+          interfaces:
+            - name: red
+              masquerade: {} <1>
+              ports:
+                - port: 80 <2>
+      networks:
+      - name: red
+        pod: {}
+      volumes:
+      - cloudInitNoCloud:
+          networkData: |
+            version: 2
+            ethernets:
+              eth0:
+                dhcp4: true
+                addresses: [ fd10:0:2::2/120 ] <3>
```

Comment:
@phoracek Shouldn't the documentation indicate (not necessarily here) that this is not the address that is used externally?
Do we want to also explain how to check this address (via vmi.status.ipAddresses)?

Reply:
>@phoracek Shouldn't the documentation indicate (not necessarily here) that this is not the address that is used externally?

@yossisegev I feel that is already said in https://github.com/openshift/openshift-docs/pull/22490/files#diff-f05ea61fe2ae34ffbbab07210c5529d5R10 - "incoming and outgoing traffic for the virtual machine is routed to both the IPv4 address and the unique IPv6 address of the virt-launcher Pod"

> Do we want to also explain how to check this address (via vmi.status.ipAddresses)?

I (personally) feel that it is reasonable to assume users will know that - but being explicit hardly hurts.

Reply:
@yossisegev and @maiqueb 
Added a new para to the intro so that this is more explicit:
The IPv6 network address must be statically set to `fd10:0:2::2/120` with a default gateway of `fd10:0:2::1` in the virtual machine configuration. These are used by the virt-launcher pod to route IPv6 traffic to the virtual machine and are not used externally.

Have also added a verification step after the procedure but wondering: is there a more precise command we could use?

Reply:
> @yossisegev and @maiqueb
> Added a new para to the intro so that this is more explicit:
> The IPv6 network address must be statically set to `fd10:0:2::2/120` with a default gateway of `fd10:0:2::1` in the virtual machine configuration. These are used by the virt-launcher pod to route IPv6 traffic to the virtual machine and are not used externally.
> 
> Have also added a verification step after the procedure but wondering: is there a more precise command we could use?

You can:
```bash
# using go-template
oc get vmi <vmi-name> -ogo-template="{{range .status.interfaces}}{{ .ipAddresses }}{{end}}"

# using jsonpath
oc get vmi <vmi-name> -ojsonpath="{.status.interfaces[*].ipAddresses}"
```

I guess the last one is more readable.

Reply:
> You can:
> 
> ```shell
> # using go-template
> oc get vmi <vmi-name> -ogo-template="{{range .status.interfaces}}{{ .ipAddresses }}{{end}}"
> 
> # using jsonpath
> oc get vmi <vmi-name> -ojsonpath="{.status.interfaces[*].ipAddresses}"
> ```
> 
> I guess the last one is more readable.

+1 
Use an explicit example, as "oc get vmi -oyaml" outputs the entire VMI manifest, and the user doesn't necessarily know where to look.
And I agree that the "ojsonpath" option is more readable.

Reply:
Thanks for the suggestions, folks. I updated the command to:
`oc get vmi <vmi-name> -o jsonpath="{.status.interfaces[*].ipAddresses}"`